File: cybersecurity/challenges/crypto-tools/8_repeated_xor/utils/divisors.py
n1 = 320
n2 = 64
n3 = 560

for i in range(2, min(n1, min(n2, n3)) + 1):
	if n1%i == 0 and n2%i == 0 and n3%i == 0:
		print(i)

File: APP/esercizi/parallel-streams/src/main/java/Divisors.java
import java.util.Collection;
import java.util.function.Function;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class Divisors implements Function<Integer, CandidateNumber>
{
    @Override
    public CandidateNumber apply(Integer n)
    {
        return new CandidateNumber(n,
                IntStream.range(1, /*Math.toIntExact(Math.round(Math.sqrt(Math.abs(n))))*/n)
                        .boxed()
                        .parallel()
                        .filter(i -> ((n % i) == 0) )
                        .collect(Collectors.toList()));
    }
}
